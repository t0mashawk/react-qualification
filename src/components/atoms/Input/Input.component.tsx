import React from 'react'

interface InputProps{
    value:string;
    onChange:(event:React.ChangeEvent<HTMLInputElement>)=>void
}
function Input(props:InputProps){
    const{value,onChange} = props
    return(
        <input value={value} onChange={ev=>onChange(ev)}></input>
    )
}

export default Input