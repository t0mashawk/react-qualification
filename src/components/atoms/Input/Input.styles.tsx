import styled from "styled-components";

export const Container = styled.div`
width: 100%;
max-width: 300px;
  > input {
   width: 100%;
   padding:5px;
  }
`;
