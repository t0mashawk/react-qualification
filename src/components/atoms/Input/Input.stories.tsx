import React from "react";
import { storiesOf } from "@storybook/react";
import Input from "./Input.component"

storiesOf("Input", module)
    .add("default",
        () => <Input value="test" onChange={()=>{}}/>
    )