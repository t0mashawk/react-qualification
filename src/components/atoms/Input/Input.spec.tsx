import React from "react";
import renderer from "react-test-renderer"

import Input from "./Input.component";

test("Input renders as expected", () => {
  const component = renderer.create(
    <Input onChange={() => null} value="Example Value" />
  );

  const tree = component.toJSON();
  
  expect(tree).toMatchSnapshot();
});