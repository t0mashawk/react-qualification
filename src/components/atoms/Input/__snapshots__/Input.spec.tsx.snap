// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders as expected 1`] = `
<div
  className="sc-AxjAm jnspPx"
>
  <input
    onChange={[Function]}
    value="Example Value"
  />
</div>
`;
